cmd/kubeadm/app/phases/certs/certs.go

```diff
@@ -51,8 +51,7 @@ func CreatePKIAssets(cfg *kubeadmapi.MasterConfiguration) error {
 		CreateAPIServerEtcdClientCertAndKeyFiles,
 	}
 
-	// Currently this is the only way we have to identify static pod etcd vs external etcd
-	if len(cfg.Etcd.Endpoints) == 0 {
+	if cfg.Etcd.Local != nil || cfg.Etcd.External == nil {
```

Comment:
`|| cfg.Etcd.External == nil`

Reply:
There is an edge case here actually when we don't know which one of the structs are defined or the other undefined (in unit tests), so hence I added this. It doesn't hurt I guess...?

Reply:
Ok. If possible this should be annotated in comments because this is the only place where this is required

Reply:
Okay, now I changed the unit tests instead, so now it works as you wanted it :+1: